genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/CommandTask.java

```diff
@@ -78,7 +79,15 @@ public void executeTask(
             );
 
             this.fts.getFile(commandSetupFile, localPath);
-            Utils.appendToWriter(writer, "source " + localPath + ";");
+
+            Utils.appendToWriter(writer, "# Sourcing setup file from command " + jobExecEnv.getCommand().getId());
+            Utils.appendToWriter(writer,
+                JobConstants.SOURCE
+                    + localPath.replace(this.jobWorkingDirectory, "${" + JobConstants.GENIE_JOB_DIR_ENV_VAR + "}")
+                    + JobConstants.SEMICOLON_SYMBOL);
+
+            // Append new line
+            Utils.appendToWriter(writer, " ");
```

Comment:
this looks like pretty much the exact same code that is in the cluster task file? Can't it be factored out into reusable?